pkg/scaling/scale_handler_raw_metrics.go

```diff
@@ -159,11 +157,10 @@ func (h *scaleHandler) sendWhenSubscribed(soName, ns, triggerName string, metric
 
 		for _, t := range targets {
 			select {
-			case <-t.done:
-				// subscriber closed
-				return
 			case t.rawMetrics <- msg:
 				// sent ok
+			case <-t.done:
+				// subscriber closed
```

Comment:
should reading from `t.done` channel also trigger closing of the channels from `t`?